### ISSUE TITLE
Sparsity Support

### ISSUE THREAD

**Issue by avik-pal** [EnzymeAD/Reactant.jl#403]
We need to compile the [SparseTensor Dialect](https://github.com/llvm/llvm-project/tree/9bd58f6f051c9b5546bbaa891fa11d3624b8d9a4/mlir/include/mlir/Dialect/SparseTensor). StableHLO already has initial support for sparse tensors

**Comment by mofeing** [comment]
yeah, current behavior is to track individual numbers so it works... well, it only works if the sparsity pattern is the same during compilation and during execution (let aside the performance implications)

**Comment by BenCurran98** [comment]
Hi, just wanted to check what the status of this issue is? It would be great to have since it will allow things like [GNNGraphs.jl](https://juliagraphs.org/GraphNeuralNetworks.jl/docs/GNNGraphs.jl/stable/) to use Reactant.jl (for deep learning on graphs). Currently one could work around this by making all graphs use dense representations when using Reactant, but would love to avoid this if possible :)
